doc/tutorials/setup_openstack_libvert.rst

```diff
@@ -0,0 +1,283 @@
+
+   ``lxc exec openstack -- su -l``
+
+9. Set up the Intel proxy inside the **openstack** container via ``/etc/environment``
+   Make sure no_proxy is properly set up in ``/etc/environment`` inside the container. Both IP addresses assigned to **eth0** and **eth1** and their subnets must be included. For example:
+
+   ``no_proxy=intel.com,.intel.com,10.0.0.0/8,192.168.0.0/16,localhost,.local,127.0.0.0/8,134.134.0.0/16``
+
+10. Add a new user named **stack** and set permissions:
+
+    ``sudo useradd -s /bin/bash -d /opt/stack -m stack``
+
+    ``echo "stack ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers``
+
+11. Log out and restart the **openstack** container:
+
+    ``lxc restart openstack``
+
+The **openstack** container is now properly configured for OpenStack. Use ``lxc list`` to verify that both **eth0** and **eth1** appear in the container.
+
+Set up ACRN prerequisites inside the container
+**********************************************
+
+1. Log in to the **openstack** container as **stack** user:
+
+   ``lxc exec openstack -- su -l stack``
+
+2. Download and compile ACRN’s source code. Refer to :ref:`getting-started-building`.
+
+   - make
+   - cd misc/acrn-manager/; make
+   - Install only the user-space components (acrn-dm/acrnctl/acrnd)
+
+3. Download, compile, and install ``iasl``. Refer to :ref:`Prepare the User VM <prepare-UOS>`.
+
+Set up libvirt
+**************
+
+1. Install the required packages:
+
+   ``sudo apt install libdevmapper-dev libnl-route-3-dev libnl-3-dev python automake autoconf autopoint libtool xsltproc libxml2-utils gettext``
+
+2. Download libvirt/ACRN:
+
+   ``git clone https://github.com/projectacrn/acrn-libvirt.git``
+
+3. Go to the libvert directory (cd libvirt) and enter the following:
+
+   ``./autogen.sh --prefix=/usr --disable-werror --with-test-suite=no --with-qemu=no --with-openvz=no --with-vmware=no --with-phyp=no --with-vbox=no --with-lxc=no --with-uml=no --with-esx=no``
+
+   ``make``
+
+   ``sudo make install``
+
+4. Edit and enable these options in ``/etc/libvirt/libvirtd.conf``:
+
+   ``unix_sock_ro_perms = "0777"``
+
+   ``unix_sock_rw_perms = "0777"``
+
+   ``unix_sock_admin_perms = "0777"``
+
+5. Run the following command:
+
+   ``sudo systemctl daemon-reload``
+
+
+Set up OpenStack
+****************
+
+Use DevStack to install OpenStack. Refer to the `DevStack instructions <https://docs.openstack.org/devstack/>`_.
+
+1. Use the latest maintenance branch **stable/train** to ensure OpenStack stability:
+
+   ``git clone https://opendev.org/openstack/devstack.git -b stable/train``
+
+2. Go to the devstack directory (cd devstack) and apply the following patch:
+
+   ``0001-devstack-installation-for-acrn.patch``
+
+3. Edit ``lib/nova_plugins/hypervisor-libvirt``:
+
+   Change ``xen_hvmloader_path`` to the location of your OVMF image file. A stock image is included in the ACRN source tree (devicemodel/bios/OVMF.fd).
+
+4. Copy the attached ``local.conf`` to ``devstack/``.
```

Comment:
Need to provide the `local.conf` file (or include its content in-line in this doc if not too long)